.circleci/config.yml

```diff
@@ -50,9 +58,74 @@ jobs:
             - "~/.m2"
             - "/root/.m2"
       - run:
-          name: Build and deploy
+          name: Build and deploy libraries
           command: |
             echo 'export PATH=$JAVA_HOME/bin:$PATH' >> $BASH_ENV
             source $BASH_ENV
             ./mvnw -s .circleci/settings.xml deploy -DskipTests -Dtycho.localArtifacts=$LOCAL_ARTIFACTS
+  deploy-service-dev:
+    <<: *env
+    steps:
+      - checkout
+      - setup_remote_docker
+      - restore_cache:
+          keys:
+            - snow-owl-mvn-cache-{{ checksum "pom.xml" }}
+            - snow-owl-mvn-cache-
+          paths:
+            - "~/.m2"
+            - "/root/.m2"
+      - run:
+          name: Build prerequisites
+          command: |
+            docker login -u "$DOCKER_USER" -p "$DOCKER_PASS" quay.io
+            apk add --no-cache make=4.2.1-r2 nodejs npm openjdk11 --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
+            npm config set unsafe-perm true
+            npm install -g npm
+      - save_cache:
+          key: snow-owl-mvn-cache-{{ checksum "pom.xml" }}
+          paths:
+            - "~/.m2"
+            - "/root/.m2"
+      - run:
+          name: Build and push docker image
+          command: |
+            echo 'export PATH=$JAVA_HOME/bin:$PATH' >> $BASH_ENV
+            source $BASH_ENV
+            make build-project
+            export SNOWOWL_RPM_PACKAGE=`find ./releng/com.b2international.snowowl.server.update/target -name "snow-owl-oss*.rpm"`
+            cp "${SNOWOWL_RPM_PACKAGE}"  ./docker/`basename "${SNOWOWL_RPM_PACKAGE}"`
+            export RELEASE_VERSION=`git describe --abbrev=0 --tags`
+            make build-docker
+            make push
+      - deploy:
+          name: Deploy dev
+          command: make deploy-dev
+  deploy-service-staging:
+    <<: *env
+    steps:
+      - checkout
+      - setup_remote_docker
+      - restore_cache:
+          keys:
+            - snow-owl-mvn-cache-{{ checksum "pom.xml" }}
+            - snow-owl-mvn-cache-
+          paths:
+            - "~/.gradle"
+            - "~/.m2"
+            - "/root/.gradle"
+      - run:
+          name: Build prerequisites
+          command: |
+            docker login -u "$DOCKER_USER" -p "$DOCKER_PASS" quay.io
+            apk add --no-cache make=4.2.1-r0 openjdk11 --repository=http://dl-cdn.alpinelinux.org/alpine/edge/community
+            make download-deps
+      - deploy:
+          name: Staging Deploy
+          command: |
+            git branch --remotes --contains HEAD | grep  "^\s*origin/master$" > /dev/null
```

Comment:
what does it do, do we need it? maybe delete this line

Reply:
removing